src/components/carousel/index.tsx

```diff
@@ -10,12 +10,14 @@ import ThumbnailPagination from './ThumbnailPagination';
 import Slide from './Slide';
 import NumbersPagination from './NumbersPagination';
 import NavigationButton from './NavigationButton';
+import DotsPagination from './DotsPagination';
 
 type SharedProps = {
   startIndex?: number;
   onSlideClick?: (index: number) => void;
   onSlideSelect?: (index: number) => void;
   withNumbersPagination?: boolean;
+  withDotsPagination?: boolean;
```

Comment:
this creates an impossible state. If you pass `withNumbersPagination=true` and `withDotsPagination=true`. I would go for something like: 
`pagination: PaginationType` instead of the booleans

Reply:
true!